test/views/sections/items/ConfirmCommodityCodeViewSpec.scala

```diff
@@ -0,0 +1,142 @@
+        }
+
+        "have a link to change commodity code" in {
+          doc.getElementById("changeItemCommodityCode1").attr("href") mustBe
+            controllers.sections.items.routes.ItemCommodityCodeController.onPageLoad(testErn, testDraftId, testIndex1, NormalMode).url
+        }
+      }
+    }
+
+    Map(testExciseProductCodeS500 -> testCommodityCodeWine,
+      testExciseProductCodeT300 -> testCommodityCodeWine,
+      testExciseProductCodeS400 -> testCommodityCodeWine,
+      testExciseProductCodeE600 -> testCommodityCodeWine,
+      testExciseProductCodeE800 -> testCommodityCodeWine,
+      testExciseProductCodeE910 -> testCommodityCodeWine).foreach { case (exciseProductCode, commodityCode) =>
+      Seq(ConfirmCommodityCodeMessages.English).foreach { messagesForLanguage =>
+
+        s"when being rendered in lang code of '${messagesForLanguage.lang.code}' for EPC code ${exciseProductCode}'" - {
+
+          implicit val msgs: Messages = messages(Seq(messagesForLanguage.lang))
+
+          implicit val request: DataRequest[AnyContentAsEmpty.type] =
+            dataRequest(FakeRequest(), emptyUserAnswers
+              .set(ItemExciseProductCodePage(testIndex1), exciseProductCode)
+              .set(ItemCommodityCodePage(testIndex1), commodityCode)
+            )
+          val view = app.injector.instanceOf[ConfirmCommodityCodeView]
+
+          implicit val doc: Document = Jsoup.parse(view(
+            controllers.sections.items.routes.ConfirmCommodityCodeController.onSubmit(testErn, testDraftId, testIndex1, NormalMode),
+            SummaryList(Seq(
+              ItemExciseProductCodeSummary.row(testIndex1),
+              ItemCommodityCodeSummary.row(testIndex1, GoodsTypeModel(exciseProductCode.code), emptyUserAnswers
+                .set(ItemExciseProductCodePage(testIndex1), exciseProductCode)
+                .set(ItemCommodityCodePage(testIndex1), commodityCode))
+            ).flatten)
+          ).toString())
+
+          behave like pageWithExpectedElementsAndMessages(Seq(
+            Selectors.title -> messagesForLanguage.title,
+            Selectors.h1 -> messagesForLanguage.heading,
+            Selectors.h2(1) -> messagesForLanguage.caption,
+            Selectors.govukSummaryListKey(1) -> messagesForLanguage.exciseProductCode,
+            Selectors.govukSummaryListKey(2) -> messagesForLanguage.commodityCode,
+            Selectors.button -> messagesForLanguage.confirmCodes,
+          ))
+
+          "have a link to change excise product code" in {
+            doc.getElementById("changeItemExciseProductCode1").attr("href") mustBe
+              controllers.sections.items.routes.ItemExciseProductCodeController.onPageLoad(testErn, testDraftId, testIndex1, NormalMode).url
+          }
+
+          "does not have a link to change commodity code" in {
+            intercept[NullPointerException] {
+              doc.getElementById("changeItemCommodityCode1").attr("href")
+            }
```

Comment:
I feel like this is a bit misleading, because a NPE could be thrown for other reasons. Not a major thing, though